### PR TITLE
Avoid allocating the list for empty results and initialize the list size with message count to prevent growing

### PR DESCRIPTION
**Pull request by danielmarbach**
Small tweak @xinchen10 

// cc @jsquire @JoshLove-msft

### CODE REVIEW

File: src/ReceivingAmqpLink.cs [hunk 1]
```diff
@@ -224,19 +224,20 @@ public IAsyncResult BeginReceiveMessages(int messageCount, TimeSpan timeout, Asy
...
+                    messages = new List<AmqpMessage>(messageCount);
                     for (int i = 0; i < messageCount && this.messageQueue.Count > 0; i++)
                     {
                         messages.Add(this.messageQueue.Dequeue());
                     }
                 }
             }
 
-            if (!messages.Any() && timeout > TimeSpan.Zero)
+            if (messages?.Count == 0 && timeout > TimeSpan.Zero)
```
Comment (xinchen10) [inline src/ReceivingAmqpLink.cs:240]: `messages?.Count` returns a nullable int and its value is not 0 when messages is null. For example, with the following code, `isEmpty` is false.
```
            List<string> list = null;
            bool isEmpty = list?.Count == 0;
```

Comment (danielmarbach) [reply src/ReceivingAmqpLink.cs:240]: Yeah will fix. I didn't have time to look at the test results yet. Thanks for coming back to me!

File: src/ReceivingAmqpLink.cs [hunk 2]
```diff
@@ -224,19 +224,20 @@ public IAsyncResult BeginReceiveMessages(int messageCount, TimeSpan timeout, Asy
...
+                    messages = new List<AmqpMessage>(messageCount);
                     for (int i = 0; i < messageCount && this.messageQueue.Count > 0; i++)
                     {
                         messages.Add(this.messageQueue.Dequeue());
                     }
                 }
             }
 
-            if (!messages.Any() && timeout > TimeSpan.Zero)
+            if ((messages?.Count ?? 0) == 0 && timeout > TimeSpan.Zero)
```
Comment (xinchen10) [inline src/ReceivingAmqpLink.cs:240]: A check for `messages == null && timeout > TimeSpan.Zero` is good enough here.

Comment (danielmarbach) [reply src/ReceivingAmqpLink.cs]: Pushed

**Comment by danielmarbach** [comment]
Thanks @xinchen10 for taking this in.

**Comment by JoshLove-msft** [comment]
Can this be merged into hotfix (or the equivalent change made there) so that we can benefit from the improvements in the next release?

**Pull request merged**